plugins/kubernetes/test/controllers/kubernetes_dashboard_controller_test.rb

```diff
@@ -1,34 +1,42 @@
 require_relative '../test_helper'
 
 describe KubernetesDashboardController do
+  let(:project) { projects(:test) }
+  let(:environment) { environments(:production) }
+  let(:release) { kubernetes_releases(:test_release) }
+
   as_a_viewer do
     before do
       Kubernetes::Cluster.any_instance.stubs(:client).returns(Kubeclient::Client.new('http://cluster.localhost'))
-      stub_request(:get, %r{http://cluster\.localhost.*}).to_return(body: '{}')
     end
 
     it 'returns an empty array if no pods are running' do
-      get :index, project_id: :foo, environment: environments(:production).id, format: :json
+      environment.cluster_deploy_groups.each do |cluster_deploy_group|
+        stub_request(:get, %r{http://cluster\.localhost.*/namespaces/#{cluster_deploy_group.namespace}.*})
+          .to_return(body: '{}')
```

Comment:
The problem is not the pattern, is the number of times you are registering the stub. I mean, a valid alternative using the original pattern would probably be:

```
stub_request(:get, %r{http://cluster\.localhost.*}).times(environment.cluster_deploy_groups.count).to_return(body: '{}')
```

The advantage of having the explicit namespace in the pattern is that, if some other request is made that does not fall into the registered patterns, the test will fail.